.github/workflows/hypersdk-ci.yml

```diff
@@ -0,0 +1,295 @@
+name: HyperSDK CI
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  mock_gen:
+    name: Auto-Generated Code Checker
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - name: Set up Go
+        uses: ./.github/actions/install-go
+      - shell: bash
+        run: scripts/mock.gen.sh
+      - shell: bash
+        run: scripts/tests.clean.sh
+
+  hypersdk-lint:
+    runs-on: ubuntu-latest
+    timeout-minutes: 10
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+      - name: Set up Go
+        uses: ./.github/actions/install-go
+      - name: Run static analysis tests
+        shell: bash
+        run: scripts/tests.lint.sh
+      - name: Run shellcheck
+        shell: bash
+        run: scripts/tests.shellcheck.sh
+
+  go_mod_tidy:
+    name: Tidy
+    runs-on: ubuntu-20.04
+    steps:
+      - uses: actions/checkout@v4
+      - name: Set up Go
+        uses: ./.github/actions/install-go
+      - shell: bash
+        run: go mod tidy
+      - shell: bash
+        working-directory: ./examples/morpheusvm
+        run: go mod tidy
+      - shell: bash
+        working-directory: ./examples/tokenvm
+        run: go mod tidy
+      - shell: bash
+        run: scripts/tests.clean.sh
+
+  hypersdk-unit-tests:
+    runs-on: ubuntu-latest
+    timeout-minutes: 10
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+      - name: Set up Go
+        uses: ./.github/actions/install-go
+      - name: Run unit tests
+        shell: bash
+        run: scripts/tests.unit.sh
+
+  finished-all-hypersdk-tests:
+    runs-on: ubuntu-latest
+    needs: [mock_gen, hypersdk-lint, go_mod_tidy, hypersdk-unit-tests]
+    steps:
+      - name: Finished all HyperSDK tests
+        run: echo "Finished all HyperSDK tests"
+
+  # TokenVM
+  tokenvm-lint:
+    runs-on: ubuntu-latest
+    needs: [finished-all-hypersdk-tests]
+    timeout-minutes: 10
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+      - name: Set up Go
+        uses: ./.github/actions/install-go
+        with:
+          cache-dependency-path: |
+            go.sum
+            examples/tokenvm/go.sum
+      - name: Run static analysis tests
+        working-directory: ./examples/tokenvm
+        shell: bash
+        run: scripts/tests.lint.sh
+      - name: Build vm, cli, faucet, feed
+        working-directory: ./examples/tokenvm
+        shell: bash
+        run: scripts/build.sh
+
+  tokenvm-unit-tests:
+    runs-on: ubuntu-latest
```

Comment:
This used to run on a labelled machine, do we need that? I'm assuming we don't since the timeout is 10 instead of 25, but easy to put back